### PR TITLE
JR2-315 Add number of candidates found on landing page

### PR DESCRIPTION
**Pull request by michael-wirth**
Please review, but PR is not finished yet

### CODE REVIEW

File: src/main/webapp/app/home/state-management/reducers/candidate-search-tool.reducers.ts [hunk 1]
```diff
@@ -3,15 +3,22 @@ import {
...
 
 export function candidateSearchToolReducer(state = initialState, action: Actions): CandidateSearchToolState {
     let newState;
     switch (action.type) {
         case CANDIDATE_SEARCH_TOOL_SUBMITTED:
+        case CANDIDATE_SEARCH_TOOL_COUNT:
             newState = Object.assign({}, action.payload);
             break;
-
+        case CANDIDATE_SEARCH_TOOL_COUNT_UPDATE:
```
Comment (birom) [inline src/main/webapp/app/home/state-management/reducers/candidate-search-tool.reducers.ts]: This article: [NgRx: Patterns and Techniques](https://blog.nrwl.io/ngrx-patterns-and-techniques-f46126e2b1e5) distingueshes command actions from event actions. According to this I propose the following naming convention:
-command actions: verb's base form (present tense)
-event actions: verb in past tense

File: src/main/webapp/app/candidate-search/services/candidate.service.ts [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
 import { Observable } from 'rxjs/Observable';
-import { Http, Response } from '@angular/http';
+import { BaseRequestOptions, Http, Response, URLSearchParams } from '@angular/http';
+import { JhiDateUtils } from 'ng-jhipster';
+import { ResponseWrapper } from '../../shared';
+import { TranslateService } from '@ngx-translate/core';
+import { CandidateSearchRequest } from './candidate-search-request';
 
 @Injectable()
 export class CandidateService {
```
Comment (birom) [inline src/main/webapp/app/candidate-search/services/candidate.service.ts]: Add tests

File: src/main/webapp/app/candidate-search/language-selector/language-selector.component.ts [hunk 3]
```diff
@@ -108,7 +108,7 @@ export class LanguageSelectorComponent implements ControlValueAccessor {
     }
 
     private _onChange = (_: any) => {
-    }
+    };
```
Comment (michael-wirth) [inline src/main/webapp/app/candidate-search/language-selector/language-selector.component.ts]: remove `;`

File: src/main/webapp/app/candidate-search/services/candidate.service.ts [hunk 4]
```diff
@@ -27,4 +34,66 @@ export class CandidateService {
...
+
+    private convertResponse(res: Response): ResponseWrapper {
+        const jsonResponse = res.json();
+        for (let i = 0; i < jsonResponse.length; i++) {
+            this.convertItemFromServer(jsonResponse[i]);
+        }
+        return new ResponseWrapper(res.headers, jsonResponse, res.status);
+    }
+
+    private convertItemFromServer(entity: any) {
```
Comment (michael-wirth) [inline src/main/webapp/app/candidate-search/services/candidate.service.ts]: this is copy-paste from job-service. cleanup or remove (if we don't have any dates in the entity)

File: src/main/webapp/app/home/state-management/actions/candidate-search-tool.actions.ts [hunk 5]
```diff
@@ -2,10 +2,26 @@ import { Action } from '@ngrx/store';
 import { CandidateSearchToolState } from '../state/candidate-search-tool.state';
 
 export const CANDIDATE_SEARCH_TOOL_SUBMITTED = 'CANDIDATE_SEARCH_TOOL_SUBMITTED';
+export const CANDIDATE_SEARCH_TOOL_COUNT_UPDATE = 'CANDIDATE_SEARCH_TOOL_COUNT_UPDATE';
+export const CANDIDATE_SEARCH_TOOL_COUNT_UPDATED = 'CANDIDATE_SEARCH_TOOL_COUNT_UPDATED';
```
Comment (michael-wirth) [inline src/main/webapp/app/home/state-management/actions/candidate-search-tool.actions.ts]: there is no Update process.

better use `CANDIDATE_SEARCH_TOOL_COUNT` or `CANDIDATE_SEARCH_TOOL_COUNTED`.
Does this match our naming conventions?

File: src/main/webapp/app/home/state-management/effects/home.effects.ts [hunk 6]
```diff
@@ -10,9 +10,13 @@ import {
     JobSearchToolSubmittedAction
 } from '../actions/job-search-tool.actions';
 import {
+    CANDIDATE_SEARCH_TOOL_COUNT_UPDATE,
     CANDIDATE_SEARCH_TOOL_SUBMITTED,
+    CandidateSearchToolCountUpdateAction,
```
Comment (michael-wirth) [inline src/main/webapp/app/home/state-management/effects/home.effects.ts]: Rename the actions according the issue above

**Pull request merged**